### PR TITLE
docs(flink): override default install instructions

### LINKED ISSUE #7738: docs(flink): override default install instructions
### Please describe the issue

Installing the Flink backend should be done with `pip install ibis-framework apache-flink` (or `pip install ibis-framework 'apache-flink>=1.18'`?). This differs from the installation instructions for every other backend, so probably needs to be overridden in the overlay.

Also, we don't yet have a path to supporting installation with `conda/mamba`, [pending Flink being made available via `conda-forge`](https://apache-flink.slack.com/archives/C065944F9M2/p1701811320307299).

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by deepyaman**
Closes #7738 

`pip`:
image

`conda`/`mamba`/`pixi`:
image

### CODE REVIEW

File: docs/_tabsets/install.qmd [hunk 1]
```diff
@@ -34,30 +36,53 @@ installers = [
...
+                print("Install alongside the `apache-flink` package:")
+                print()
+                print(f"```bash\npip install ibis-framework apache-flink\n```")
+
+            else:
+                print(
+                    dedent(
+                        """\
+                        ::: {.callout-important}
+                        PyFlink is not available on conda-forge; please
```
Comment (cpcloud) [inline docs/_tabsets/install.qmd:66]:
```suggestion
                        ## PyFlink is not available on conda-forge; please
```

File: docs/_tabsets/install.qmd [hunk 2]
```diff
@@ -34,30 +36,53 @@ installers = [
...
+                        use `pip` to install the PyFlink backend instead.
+                        :::\
+                        """
```
Comment (cpcloud) [inline docs/_tabsets/install.qmd:69]:
```suggestion
                        :::"""
```

**Comment by cpcloud** [comment]
Do you need to rebase to pick up the `flink.qmd` doc? Otherwise the link checker will continue to fail because it won't allow broken links :)

**Comment by deepyaman** [comment]
> Do you need to rebase to pick up the `flink.qmd` doc? Otherwise the link checker will continue to fail because it won't allow broken links :)

Done!

File: docs/_tabsets/install.qmd [hunk 3]
```diff
@@ -34,30 +36,53 @@ installers = [
...
-        extra = backend.get("extra", mod)
 
-        print(f"## {name}")
+        print(f"## {backend_name}")
         print()
-        print(line.format(extra=extra))
-        print()
-        print(f"```bash\n{cmd.format(extra=extra)}\n```")
+
+        if backend_name == "Flink":
```
[the remark this answers is not in the document]
Comment (deepyaman) [reply docs/_tabsets/install.qmd:55]: I think some of these aspects should be resolvable in the shorter term. We can try once more to see if the (Py)Flink maintainers would be open to publishing to conda-forge; if not, we could add the feedstock ourselves? Let me reach out once more.

Regarding the `pip install`, the way to make it more transparent for the user would be to do what you and @gforsyth mentioned, in terms of overriding the PyArrow bounds (as previously done for Snowflake, apparently).

[the remark this answers is not in the document]
Comment (gforsyth) [reply docs/_tabsets/install.qmd:55]: I took a quick lock at getting flink on `conda-forge`.

Woof.
We need to add feedstocks for `pemja` and `apache-flink-libraries` first.

**Review by cpcloud — APPROVED** [review]

**Pull request merged**